Addressables new

JesterX666 opened the pull request:
The basics of Addressable Sounds / Music.
Includes a revamp of SoundManager, Sound Messages, Sound Spawn

Also, sound spawn is now working with a unique token that identifies an INSTANCE of a sound playing.

This is NOT READY to be merged with Develop yet.  But a code-review would be nice.  Many many tests are still needed to be considered "mergeable" with Develop.

Job remaining to be done:

All sounds need to be reattached to all components.  There is a special "comment" : //JESTER that indicates where that job needs to be done.  Some place is harder than others (ex:  Footstep logic has to be moved from the OLD SoundManager to somewhere else).

Please leave me your comments

JesterX666 committed:
Music

JesterX666 committed:
Update Assets.asmdef

JesterX666 committed:
Some converted Game Objects

JesterX666 committed:
Update SoundManager.cs

JesterX666 committed:
Create GameObjectToPrefab.cs

JesterX666 committed:
GameObjectToPrefab

JesterX666 committed:
After Conversion

JesterX666 committed:
Prefabs Moved

JesterX666 committed:
Sound Prefab to Addressables

JesterX666 committed:
Reclassification of musics

JesterX666 committed:
All sounds reattached to SoundManager

JesterX666 committed:
Temp

JesterX666 committed:
Latest adjustements to SoundManager

JesterX666 committed:
Deactivate All Hardcoded sounds

JesterX666 committed:
After Merge

JesterX666 committed:
Delete Firebase.meta

JesterX666 committed:
New Addressable Groups + Tools

JesterX666 committed:
Prefabs Created

JesterX666 committed:
Addressables Creation

JesterX666 committed:
Add // JESTER comments.

JesterX666 committed:
Sounds progressing well

JesterX666 committed:
Merge remote-tracking branch 'upstream/Addressables' into Addressables-New

JesterX666 committed:
Some sounds re-added

JesterX666 committed:
Merge remote-tracking branch 'origin/Addressables' into Addressables-New

JesterX666 committed:
Merge

JesterX666 committed:
Syntax

JesterX666 committed:
Updates

JesterX666 committed:
Merge from Upstream/Addressables

JesterX666 committed:
Updates

JesterX666 committed:
Merge remote-tracking branch 'upstream/Addressables' into Addressables-New

JesterX666 committed:
Updates

JesterX666 committed:
Updates

JesterX666 committed:
Updates

JesterX666 committed:
Prefabs

JesterX666 committed:
Update BVOld_Microwave.prefab

Bod9001 commented on UnityProject/Assets/Scripts/Addressables/Types/AddressableReference.cs:
```diff
@@ -1,72 +1,45 @@
 		}
-
-
-		private async Task<T> LoadAsset()
```
It seems you accidentally reverted this to an old to an old estate

Bod9001 commented on UnityProject/Assets/Scripts/Closets/InteractableFireCabinet.cs:
```diff
@@ -141,7 +142,8 @@ private void Open()
 		IsClosed = true;
-		SoundManager.PlayAtPosition("OpenClose", transform.position, gameObject);
+		// JESTER
+		//SoundManager.PlayAtPosition("OpenClose", transform.position, gameObject);
```
still have a few bits and bobs to work on

Bod9001 commented on UnityProject/Assets/Scripts/Editor/Tools/PrefabToAddressable.cs:
```diff
@@ -0,0 +1,132 @@
+/// <summary>
+/// Will convert all prefabs in a path to addressable and put them in chosen Addressable Group with chosen labels.
+/// </summary>
+public class PrefabToAddressable : EditorWindow
```
thanks for the tools

JesterX666 committed:
Step Sounds first commit

JesterX666 committed:
Step Sounds

JesterX666 committed:
StepSounds

JesterX666 committed:
Revert Player_V3

Revert

JesterX666 committed:
Merge from Addressables

JesterX666 committed:
After updating branch from Addressables.

JesterX666 committed:
Better finding of missing addressable

Bod9001 commented on UnityProject/Assets/Scripts/Editor/Tools/GameObjectToPrefab.cs:
```diff
@@ -0,0 +1,122 @@
+
+	void OnGUI()
+	{
+		EditorGUILayout.HelpBox("Choose the GameObject that contains the childs you want to convert to prefabs", MessageType.Info); ;
```
Codacy found an issue: [Remove this empty statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Accidental Double ; 
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/UI/Objects/Commons/Jukebox/GUI_Jukebox.cs:
```diff
@@ -31,11 +31,11 @@ public class GUI_Jukebox : NetTab
 		}
 
-		public void OnTabOpenedHandler(ConnectedPlayer connectedPlayer)
+		public async void OnTabOpenedHandler(ConnectedPlayer connectedPlayer)
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Player/PlayerSync.Server.cs:
```diff
@@ -1,982 +1,1038 @@
+                );
+            }
+
+            step = !step;
```
Codacy found an issue: [Make the enclosing instance method 'static' or remove this set on the 'static' field.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Editor/Tools/AddressableToGameObjectTree.cs:
```diff
@@ -0,0 +1,128 @@
+
+			EditorGUILayout.Separator();
+
+			EditorGUILayout.HelpBox("Choose the GameObject that contains the tree of Asset References", MessageType.Info); ;
```
Codacy found an issue: [Remove this empty statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Accidental Double ;
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -379,72 +365,94 @@ public static void PlayNetworkedAtPos(string sndName, Vector3 worldPos, AudioSou
+	/// <remarks>
+	///		If Global is true, the sound may still be muffled if the source is configured with the muffled mixer.
+	/// </remarks>
+	public static async void Play(AddressableAudioSource addressableAudioSource, string soundSpawnToken, bool polyphonic = false, bool global = true)
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Editor/Addressable/AddressableReferencePropertyDrawer.cs:
```diff
@@ -0,0 +1,204 @@
+using System.Reflection;
+using AddressableReferences;
+using UnityEditor;
+using UnityEditor.AddressableAssets;
```
Codacy found an issue: [Change this constant to a 'static' read-only property.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
The solution suggested by Codacy wasn't adequate.  Made the property private instead.
Will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -83,156 +103,87 @@ private void Init()
 	/// </summary>
-	public static void PlayNetworked(string sndName, float pitch = -1,
+	/// <param name="addressableAudioSources">List of sounds to be played.  If more than one sound is specified, one will be picked at random</param>
+	public static async void PlayNetworked(List<AddressableAudioSource> addressableAudioSources, float pitch = -1,
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Addressables/Types/AddressableReference.cs:
```diff
@@ -1,72 +1,46 @@
 		{
+			if ((AssetReference == null) && (!string.IsNullOrWhiteSpace(Path)))
+			{
+				return !(Asset == null);
```
Codacy found an issue: [Use the opposite operator ('!=') instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -313,15 +295,18 @@ public static void PlayNetworkedAtPos(string sndName, Vector3 worldPos, AudioSou
+	/// <param name="recipient">The player that will receive the sound</param>
+	/// <param name="addressableAudioSources">The sound to be played.  If more than one is specified, one will be picked at random.</param>
+	/// <param name="pitch">The pitch variation of the sound.  Null for default pitch.</param>
+	public static async void PlayNetworkedForPlayer(GameObject recipient, List<AddressableAudioSource> addressableAudioSources, float? pitch = null,
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -379,72 +365,94 @@ public static void PlayNetworkedAtPos(string sndName, Vector3 worldPos, AudioSou
 	/// <param name="audioSourceParameters">Parameters for how to play the sound</param>
 	/// <param name="polyphonic">Should the sound be played polyphonically</param>
-	public static void Play(string name, AudioSourceParameters audioSourceParameters, bool polyphonic = false)
+	public static async void Play(AddressableAudioSource addressableAudioSource, string soundSpawnToken, AudioSourceParameters audioSourceParameters, bool polyphonic = false)
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/UI/Jukebox/GUI_Jukebox.cs:
```diff
@@ -0,0 +1,68 @@
+		}
+	}
+
+	public async void OnTabOpenedHandler(ConnectedPlayer connectedPlayer)
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
That class should not exist anymore it's moved into UI/Object.  Why is it still there?
Deleted on my branch.
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -379,72 +365,94 @@ public static void PlayNetworkedAtPos(string sndName, Vector3 worldPos, AudioSou
+		ApplyAudioSourceParameters(audioSourceParameters, soundSpawn);
 
-		Instance.PlaySource(sound, polyphonic, true, audioSourceParameters != null && audioSourceParameters.MixerType != MixerType.Unspecified);
+		Instance.PlaySource(soundSpawn, polyphonic, true, audioSourceParameters != null && audioSourceParameters.MixerType != MixerType.Unspecified);
```
Codacy found an issue: [Remove this default value assigned to parameter 'Global'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Not sure I like this syntax.  It forces us to pass parameters in non-standard order like such:

		Instance.PlaySource(soundSpawn, polyphonic, forceMixer: audioSourceParameters != null && audioSourceParameters.MixerType != MixerType.Unspecified);

Never been a fan of unordered/skipping parameters.

Fixed it anyway.
Will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -83,156 +103,87 @@ private void Init()
-		source.isPlaying = true;
-		CopySource(source.audioSource, sourceToCopy);
-		return source;
+		return await EnsureAddressableAudioSourceFromCache(addressableAudioSources);
```
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 commented on UnityProject/Assets/Scripts/Objects/Machines/Jukebox.cs:
```diff
@@ -1,285 +1,305 @@
+		//}
+	}
+
+	public async void Play()
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Fixed.  Will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Editor/Tools/FindAddressableAudioSource.cs:
```diff
@@ -0,0 +1,131 @@
+
+		void OnGUI()
+		{
+			EditorGUILayout.HelpBox("Choose the name of the type you want to find\n\nYou can use the syntax described in .NET Type.GetType(string) documentation\n\nEx.: AddressableReferences.AddressableAudioSource,Assets for the type AddressableAudioSource, in namespace AddressableReferences in assembly Assets.", MessageType.Info); ;
```
Codacy found an issue: [Remove this empty statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

JesterX666 replied:
Fixed, will commit soon.

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -256,42 +207,73 @@ private string[] GetMatchingSounds(string pattern)
+			bool polyphonic = false, bool Global = true, GameObject sourceObj = null, ShakeParameters shakeParameters = null)
 	{
-		sndName = Instance.ResolveSoundPattern(sndName);
+		AddressableAudioSource addressableAudioSource = await GetAddressableAudioSourceFromCache(addressableAudioSources);
```
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -505,50 +511,64 @@ public static void GlassknockAtPosition(Vector3 worldPos, GameObject performer =
-	public static void PlayAtPosition(string name, Vector3 worldPos, bool polyphonic = false,
+	/// <param name="addressableAudioSources">Sound to be played.  If more than one is specified, one will be picked at random.</param>
+	/// <param name="soundSpawnToken">The token that identifies the SoundSpawn uniquely among the server and all clients </param>
+	public static async void PlayAtPosition(List<AddressableAudioSource> addressableAudioSources, string soundSpawnToken, Vector3 worldPos, bool polyphonic = false,
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 commented on UnityProject/Assets/Scripts/Objects/Machines/Jukebox.cs:
```diff
@@ -1,285 +1,305 @@
-				spriteHandler.SetSpriteSO(SpriteIdle);
-			else
-				spriteHandler.SetSpriteSO(SpriteDamaged);
+	private async void CheckSongFinished()
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -505,50 +511,64 @@ public static void GlassknockAtPosition(Vector3 worldPos, GameObject performer =
+	/// </summary>
+	/// <param name="addressableAudioSources">Sound to be played.</param>
+	/// <param name="soundSpawnToken">The SoundSpawn Token that identifies the same sound spawn instance across server and clients</returns>
+	public static async void PlayAtPosition(AddressableAudioSource addressableAudioSource, string soundSpawnToken, Vector3 worldPos, bool polyphonic = false,
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 commented on UnityProject/Assets/Scripts/Objects/Machines/Jukebox.cs:
```diff
@@ -1,285 +1,305 @@
+	{
+		List<ElementValue> valuesToSend = new List<ElementValue>();
+		valuesToSend.Add(new ElementValue() { Id = "TextTrack", Value = Encoding.UTF8.GetBytes(TrackPosition) });
+		valuesToSend.Add(new ElementValue() { Id = "TextSong", Value = Encoding.UTF8.GetBytes(await GetSongNameAsync()) });
```
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -379,72 +365,94 @@ public static void PlayNetworkedAtPos(string sndName, Vector3 worldPos, AudioSou
+	/// <param name="soundSpawnToken">The SoundSpawn Token that identifies the same sound spawn instance across server and clients</returns>
+	/// <param name="audioSourceParameters">Parameters for how to play the sound</param>
+	/// <param name="polyphonic">Should the sound be played polyphonically</param>
+	public static async void Play(List<AddressableAudioSource> addressableAudioSources, string soundSpawnToken, AudioSourceParameters audioSourceParameters, bool polyphonic = false)
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 commented on UnityProject/Assets/Scripts/Objects/PushPull.cs:
```diff
@@ -1006,126 +950,102 @@ private void OnServerTileReached(Vector3Int newPos)
 		pushPrediction = PushState.Finished;
-		switch (pushApproval)
-		{
+		switch ( pushApproval ) {
```
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 commented on UnityProject/Assets/Scripts/Objects/Machines/Jukebox.cs:
```diff
@@ -1,285 +1,305 @@
+		List<ElementValue> valuesToSend = new List<ElementValue>();
+		valuesToSend.Add(new ElementValue() { Id = "TextTrack", Value = Encoding.UTF8.GetBytes(TrackPosition) });
+		valuesToSend.Add(new ElementValue() { Id = "TextSong", Value = Encoding.UTF8.GetBytes(await GetSongNameAsync()) });
+		valuesToSend.Add(new ElementValue() { Id = "TextArtist", Value = Encoding.UTF8.GetBytes(await GetArtistNameAsync()) });
```
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -337,24 +322,25 @@ public static void PlayNetworkedAtPos(string sndName, Vector3 worldPos, AudioSou
 	/// </summary>
-	public static void PlayNetworkedForPlayerAtPos(GameObject recipient, Vector3 worldPos, string sndName,
+	/// <param name="addressableAudioSources">The sound to be played.  If more than one is specified, one will be picked at random.</param>
+	public static async void PlayNetworkedForPlayerAtPos(GameObject recipient, Vector3 worldPos, List<AddressableAudioSource> addressableAudioSources,
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 commented on UnityProject/Assets/Scripts/Managers/SoundManager.cs:
```diff
@@ -379,72 +365,94 @@ public static void PlayNetworkedAtPos(string sndName, Vector3 worldPos, AudioSou
 	///		If Global is true, the sound may still be muffled if the source is configured with the muffled mixer.
 	/// </remarks>
-	public static void Play(string name, bool polyphonic = false, bool global = true)
+	public static async void Play(List<AddressableAudioSource> addressableAudioSources, string soundSpawnToken, bool polyphonic = false, bool global = true)
```
Codacy found an issue: [Return 'Task' instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)

Bod9001 committed:
, for prefab conversion search for // JESTE_R , Still some code conversion left to do

Did a good portion of the code conversion

JesterX666 committed:
Corrections of some Codacy code smells

JesterX666 committed:
More corrections for Codacy issues

Bod9001 committed:
Done some more code conversion

Bod9001 committed:
Did the could push on for the ones that Went  to tricky

There's still a few left to do, that a bit more complicated but that's pretty much all of them, Just have to fill out the fields in the prefab now

Bod9001 committed:
added Sounds SO

Bod9001 committed:
added A few sounds prefab

Bod9001 committed:
Sounds actually work now

Bod9001 committed:
A few changes
